vpr/src/device/rr_graph_view.h

```diff
@@ -258,7 +258,13 @@ class RRGraphView {
      * This API is very powerful and developers should not use it unless it is necessary, 
      * e.g the node type is unknown. If the node type is known, the more specific routines, `node_pin_num()`, 
      * `node_track_num()`and `node_class_num()`, for different types of nodes should be used.*/
+    /** @brief Return detailed routing segment information with a given id* @note The routing segments here may not be exactly same as those defined in architecture file. They have been
+     * adapted to fit the context of routing resource graphs.
+     */
 
+    inline const t_segment_inf& rr_segments(RRSegmentId seg_id) const {
```

Comment:
From @vaughnbetz: in future, we should think about more high-level API, e.g., 
``seg_timing()``
``seg_delay()``
``seg_length()``